### PR TITLE
iMLCA and updated MLCA

### PR DESCRIPTION
**Pull request by beyeler**
Final versions of iMLCA and MLCA.

### CODE REVIEW

**Comment by islerfab** [comment]
These are a lot of files to review :-) Could you make my life a little bit easier by rebasing on development, or merging development into this branch?
Thanks!

**Comment by beyeler** [comment]
> 
> 
> These are a lot of files to review :-) Could you make my life a little bit easier by rebasing on development, or merging development into this branch?
> Thanks!

Sure. I just merged development into this branch. And yes, there are many changes because this PR includes iMLCA which was not released in the first mechlib version but was already partially reviewd by you. And I apologize for any changes only related to the formating of my IDE.

**Review by islerfab — APPROVED** [review]
I am not involved enough (anymore) in this and your research to usefully review the code in a functional sense. Thus, I skimmed through, tried to make sense of what I understood, and cleaned up a bit on the way (mostly typos), which I committed directly to this branch (I hope you agree).
So in terms of "Files changed", this PR has my approval (I liked the formatting that your IDE did, no worries, it seems to align with what mine does).

There's one thing that is bugging me, though, but I don't know if we can do something about this. Of course, you're introducing a huge line of work to this repository, but 142 commits seem like such a huge number (we have 92 so far), and I have the impression these changes could be represented in around 20-50 commits when looking at what changed in the files and the commit history. It's natural that the work of more than one year grows in # of commits, so don't take this as a criticism, please.

But still... Is there any chance you could go through the commit history by an interactive rebase and squash some of the commits together in order to clean up a little bit before merging it in? Or would that cause troubles in the versioning, since you may have referred to a certain commit/tag of this full branch at some point? If that's the case, one way around this would be to copy this branch, keep the copy with all commits in the repository, and squash segments of this one before merging it in.

I'm ok with getting all commits into `develop`, but this seems like the right time to at least think about maintaining a clean history before merging it in (I would never mess with history on the `develop` or `master` branch, of course).

What's your take on that?

File: src/main/java/org/marketdesignresearch/mechlib/mechanism/auctions/mlca/refinement/validator/ValidatorException.java
Comment (beyeler) [reply src/main/java/org/marketdesignresearch/mechlib/mechanism/auctions/mlca/refinement/validator/ValidatorException.java]: +1
I updated the serialVersionUID

**Comment by beyeler** [comment]
I agree that it would be nice to have a clean commit history. However this is not trivial due to the development path. To be on the safe side I will use a merge commit here.

**Pull request merged**
